src/genetics.rs

```diff
@@ -32,7 +31,8 @@ const MUTATION_RATE: u8 = 5; // Percent
 #[inline]
 #[must_use]
 pub fn generate_initial_population<const N: usize, const M: usize>() -> ArrayVec<Board<N>, M> {
-    let range = Uniform::from(1..=(N as u8));
+    let max_digit = u8::try_from(N).expect("digit size exceeds 255");
```

Comment:
I like the extra verbosity here to make it clear if `N` is too big :+1:

Reply:
Not just extra verbosity. There was a Clippy warning here, because Rust's `as` cast doesn't panic when `N` is out of range, it just silently truncates to 8 bits (257 becomes 1, for example). I think this is a huge Rust misfeature, but there you are.

I tried to get the assertion to happen at compile time rather than runtime, but couldn't work it out. Should probably look at it again: it can happen at compile time with the `static_assertions` crate if I can spot where the value of `N` is actually introduced.